### PR TITLE
Main parameter as an array

### PR DESCRIPTION
**Pull request by sergkudinov**
It consists changes in the parse and stringify function, and it is added a normalisation for the main parameter.

### CODE REVIEW

File: src/index.coffee.md [hunk 1]
```diff
@@ -9,7 +9,6 @@ Options are defined at the "config" level or for each command.
 
 ## About main
 
-Main is what's left after the options. Like options, "main" is
```
Comment (wdavidw) [inline src/index.coffee.md]: Was the line removed by mistake ?

File: src/index.coffee.md [hunk 2]
```diff
@@ -80,6 +84,7 @@ Parameters are defined with the following properties:
       config.description ?= 'No description yet'
       config.shortcuts = {}
       config.strict ?= false
+      sanitize_main @config
```
Comment (wdavidw) [inline src/index.coffee.md]: Can we be consistent with the lines below and use `config` instead of `@config`

File: src/index.coffee.md [hunk 3]
```diff
@@ -337,8 +342,10 @@ Convert a parameters object to an arguments array.
         if config.main
           value = lparams[config.main.name]
           throw Error "Required main argument \"#{config.main.name}\"" if config.main.required and not value?
-          keys[config.main.name] = value
-          argv.push value if value?
+          if value?
```
Comment (wdavidw) [inline src/index.coffee.md]: Do we have a test for this error ?

File: src/index.coffee.md (hunk 1 above)
Comment (sergkudinov) [reply src/index.coffee.md]: Yes, I already saw it before, but as I remember it was fine. Didn't noticed when it happened again.

Comment (wdavidw) [reply src/index.coffee.md]: Please rephrase the sentence, it not completly exact, it shall say something like "after the option and the commands have been extracted"

File: src/index.coffee.md (hunk 3 above)
Comment (sergkudinov) [reply src/index.coffee.md]: No, will add

File: src/index.coffee.md (hunk 2 above)
Comment (sergkudinov) [reply src/index.coffee.md]: Yes. will change

File: src/index.coffee.md (hunk 1 above)
Comment (sergkudinov) [reply src/index.coffee.md]: Will do tomorrow

**Pull request closed without merging**